src/App/Platforms/Android/Autofill/Fido2MakeCredentialUserInterface.cs

```diff
@@ -95,11 +98,35 @@ private async Task<(string CipherId, bool UserVerified)> CreateNewLoginForFido2C
 
         public async Task EnsureUnlockedVaultAsync()
         {
-            if (!await _stateService.IsAuthenticatedAsync() || await _vaultTimeoutService.IsLockedAsync())
+            if (!await _stateService.IsAuthenticatedAsync()
+                ||
+                await _vaultTimeoutService.IsLoggedOutByTimeoutAsync()
+                ||
+                await _vaultTimeoutService.ShouldLogOutByTimeoutAsync())
+            {
+                await NavigateAndWaitForUnlockAsync(Bit.Core.Enums.NavigationTarget.HomeLogin);
+                return;
+            }
+
+            if (!await _vaultTimeoutService.IsLockedAsync())
             {
-                // this should never happen but just in case.
-                throw new InvalidOperationException("Not authed or vault locked");
+                return;
             }
+
+            await NavigateAndWaitForUnlockAsync(Bit.Core.Enums.NavigationTarget.Lock);
+        }
+
+        private async Task NavigateAndWaitForUnlockAsync(Bit.Core.Enums.NavigationTarget navTarget)
+        {
+            _unlockVaultTcs?.TrySetCanceled();
+            _unlockVaultTcs = new TaskCompletionSource<bool>();
+            var messagingService = ServiceContainer.Resolve<IMessagingService>("messagingService");
```

Comment:
⛏️ No need for the string key for resolve; also either inject the service on the constructor or add a field with a `LazyResolve` so we don't have to resolve each time this method is called.
```suggestion
            var messagingService = ServiceContainer.Resolve<IMessagingService>();
```

Reply:
Adding as DI